Framework/Core/src/DataProcessingDevice.cxx

```diff
@@ -1155,26 +1206,62 @@ bool DataProcessingDevice::tryDispatchComputation(DataProcessorContext& context,
   };
 
   //
+  // the number of parts per input depends on the number of added PartRef object
+  // and the number of payloads in these objects. ProcessingContext provdes data
+  // through the input span where every payload is provided through a DataRef
+  // object. The part index within the input needs to be translated into a 2 dim
+  // index to access the payload message i from PartRef object j and we generate
+  // a consecutive map of these tuples
+  std::vector<std::vector<std::pair<size_t, size_t>>> partRefMaps;
```

Comment:
Can we use a proper struct, rather than a `std::pair<size_t, size_t>`?

Reply:
Fixed